crates/host/src/lib.rs

```diff
@@ -63,34 +67,59 @@ impl<'a> TryFrom<&'a str> for ResourceRef<'a> {
                         .strip_prefix("bindle://")
                         .map(Self::Bindle)
                         .context("invalid Bindle reference"),
+                    "oci" => s
+                        .strip_prefix("oci://")
+                        .map(Self::Oci)
+                        .context("invalid OCI reference"),
                     // TODO: Support other schemes
                     scheme => bail!("unsupported scheme `{scheme}`"),
                 }
             }
-            Err(url::ParseError::RelativeUrlWithoutBase) => Ok(Self::Oci(s)), // TODO: Validate
+            Err(url::ParseError::RelativeUrlWithoutBase) => {
+                match Url::parse(&format!("oci://{s}")) {
+                    Ok(_url) => Ok(Self::Oci(s)),
+                    Err(e) => Err(anyhow!(e).context("failed to parse reference as OCI reference")),
+                }
+            }
             Err(e) => {
                 bail!(anyhow!(e).context(format!("failed to parse reference `{s}`")))
             }
         }
     }
 }
 
+fn extract_server<'a>(resource_ref: &ResourceRef<'a>) -> Option<&'a str> {
```

Comment:
Why can't we use `authority` from `Url` for this?

Reply:
Hmm, `ResourceRef` contains a `&str`, which most of the time won't be a valid URL, since OCI and Bindle references don't contain a scheme. I could change `ResourceRef` to store a `Url` instead, with the type prefix as a pseudo-scheme, but then I would need to strip the scheme before passing an `impl AsRef<str>` to `fetch_actor`/`fetch_provider`, since the resource string we give to the underlying OCI/bindle clients shouldn't have them.

I think it's simpler to pull off the first part of the string before a `/`, since that defines the server for OCI and Bindle references. In fact, that's [how the OCI crate does it](https://github.com/krustlet/oci-distribution/blob/main/src/reference.rs#L264)